### PR TITLE
INFRA-6465 - Add support for setting attributes on SNS subscriptions to boto3_sns modules

### PR DESCRIPTION
**Pull request by tkwilliams**
- Add ability to set FilterPolicies (and incidentally any other modifiable attributes) on SNS subscriptions.

### CODE REVIEW

**Review by rallytime — CHANGES_REQUESTED** [review]
@tkwilliams This is failing lint. Can you fix those please?

https://jenkinsci.saltstack.com/job/pr-lint/job/PR-49352/1/warnings52Result/

**Comment by tkwilliams** [comment]
@rallytime  -- Apologies :-/  I have NO CLUE what this "Incompatible Python 3 code" pylint is fretting about might be.  I /thought/ it was a "needs two newlines" kinda thing, because it often is something arbitrary of that sort, but I added a blank line and it didn't help.  Pylint often seems overly intrusive and obtuse about perfectly good code...

**Comment by rallytime** [comment]
@tkwilliams Ohh, I didn't see it was the Python 3 error. Those hints are trickier to find. Basically you need to add the range import from six. Like this:
```
13:13:36 [salt linting]  # Import Salt libs
13:13:36 [salt linting]  import salt.utils.versions
13:13:36 [salt linting] +from salt.ext.six.moves import range
```

Pro Tip: you can find that information in the `Console Output` link on the left sidebar in the future. 😄

**Comment by tkwilliams** [comment]
@rallytime  Checking back in - is there anything I missed on this?  I can't see any of the test failures are related to my changeset but in case I missed something, please let me know :)

Thanks!

File: salt/states/boto3_sns.py [hunk 1]
```diff
@@ -158,7 +172,7 @@ def topic_present(name, subscriptions=None, attributes=None,
             ret['comment'] += '  Attribute {0} would be updated on topic {1}.'.format(attr, TopicArn)
             ret['result'] = None
             continue
-        want_val = want_val if isinstance(want_val, six.string_types) else salt.utils.json.dumps(want_val)
```
Comment (rallytime) [inline salt/states/boto3_sns.py]: Why are you removing the `salt.utils` here? We shouldn't be changing this.

File: salt/states/boto3_sns.py [hunk 2]
```diff
@@ -311,11 +378,11 @@ def topic_absent(name, unsubscribe=False, region=None, key=None, keyid=None, pro
 
 def _json_objs_equal(left, right):
     left = __utils__['boto3.ordered'](
-        salt.utils.json.loads(left)
```
Comment (rallytime) [inline salt/states/boto3_sns.py]: Same question here.

File: salt/states/boto3_sns.py [hunk 3]
```diff
@@ -311,11 +378,11 @@ def topic_absent(name, unsubscribe=False, region=None, key=None, keyid=None, pro
...
+        json.loads(left)
         if isinstance(left, six.string_types)
         else left)
     right = __utils__['boto3.ordered'](
-        salt.utils.json.loads(right)
+        json.loads(right)
```
Comment (rallytime) [inline salt/states/boto3_sns.py]: And here as well.

**Review by rallytime — CHANGES_REQUESTED** [review]
I have some requests here.

File: salt/states/boto3_sns.py (hunk 2 above)
Comment (tkwilliams) [reply salt/states/boto3_sns.py]: Ah, because of this at the top :)
```
import salt.utils.json as json
```

Comment (tkwilliams) [reply salt/states/boto3_sns.py]: It seemed a cleaner style to do it that way, but if it's counter to the norm I can revert it of course :)

**Comment by tkwilliams** [comment]
Sorry, I had replied inline on one of your questions but it may not have shown up without clicking to expand the note.  As mentioned there, it seemed a cleaner style to do it that way (by re-naming the import), but if it's counter to the norm I'm happy to switch it back....

Done :)

**Comment by rallytime** [comment]
Thank you so much @tkwilliams!

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
The lint errors are not related here, so we can just see how the rest of the tests go. Should be alright. :)

**Review by cachedout — APPROVED** [review]

**Pull request merged**